### ISSUE TITLE
NotSupportedException in PdfDocumentBuilder.Build()

### ISSUE THREAD

**Issue by cremor** [UglyToad/PdfPig#437]
I found two PDF files that cause the following exception in `PdfDocumentBuilder.Build()` after they are copied via `PdfDocumentBuilder.AddPage(source, pageNum)` to a new PDF:

```
System.NotSupportedException
  HResult=0x80131515
  Message=Object numbers must form a contiguous range
  Source=UglyToad.PdfPig
  StackTrace:
   at UglyToad.PdfPig.Writer.TokenWriter.WriteCrossReferenceTable(IReadOnlyDictionary`2 objectOffsets, IndirectReference catalogToken, Stream outputStream, Nullable`1 documentInformationReference)
   at UglyToad.PdfPig.Writer.PdfStreamWriter.CompletePdf(IndirectReferenceToken catalogReference, IndirectReferenceToken documentInformationReference)
   at UglyToad.PdfPig.Writer.PdfDocumentBuilder.CompleteDocument()
   at UglyToad.PdfPig.Writer.PdfDocumentBuilder.Build()
```

The files do **not** show any error when they are opened with Adobe Reader or other PDF readers. So I'm not sure if they are corrupted or if this is a bug in PdfPig.
Either way, can PdfPig be changed to handle this case?

The two PDF files should not be shared publicly, I'll send them to you via email.

**Comment by EliotJones** [comment]
Hi @cremor thanks for providing the files, I have pushed a change I believe fixes this to be included in tonight's package.

**Comment by EliotJones** [comment]
Closing this since the fix is available, let me know if you encounter any issues.

**Comment by cremor** [comment]
@EliotJones The creation of the PDF now works without errors, but Adobe Reader shows an error when opening a PDF that contains both source PDFs that I sent to you. The error is "There was an error opening this document. The root object is missing or invalid."

Other PDF readers I've tried can show the PDF, it's just Adobe Reader which can't. But it would still be good if that could be fixed since Adobe Reader is still widely used.

I'll sent you the created PDF via email.

**Comment by cremor** [comment]
I can confirm that this is fixed in 0.1.6-alpha-20220425-2576c, thanks!
